projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-site-settings-endpoint.php

```diff
@@ -270,6 +270,48 @@ protected function get_cast_option_value_or_null( $option_name, $cast_callable )
 		return call_user_func( $cast_callable, $option_value );
 	}
 
+	/**
+	 * Used for the WooCommmerce address options. This function returns the option value if it exists, else it looks
+	 * for a default address on the domain contact information, if it exists. Otherwise, it returns an empty string.
+	 *
+	 * @param string $key The option key to retrieve a value.
+	 *
+	 * @return string The requested address part value or an empty string.
+	 */
+	protected function get_woocommerce_address( $key ) {
```

Comment:
Another option here is to use variable functions instead of a switch statement. Then we could do something like this:

```'woocommerce_store_address' => $this->get_address( 'woocommerce_store_address', 'get_address_1' ),```

```
protected function get_address( $key, $address_method ) {
...
   $address = Domain_Contact_Information_Mapper::find_by_user_id( get_current_user_id() );
   $address_part = $address->$address_method();
...
```

It would make the function more generic, but I'm not sure that's necessary. In fact we may need specific formatting for WooCommerce. Such as doing something like Country + State.

Reply:
Nifty! Let's keep it as is for now, feels a bit easier to grok.